### ISSUE TITLE
[BUG] App does not show any of my physical or virtual devices

### ISSUE THREAD

**Issue by PeterKawa** [Baldhor/Homey-ESPhome-Enhanced#69]
Hi Baldhor,
Today I continued experimenting with ESPHome app again.

**Describe the bug**
It can't connect to my 5 existing physical devices. They are online
![Screenshot from 2023-10-22 19-18-13](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/74005072/4fb81272-65a2-4993-853d-16343686a761)

It can't connect to my 4 Homey virtual devices (doesn't show any). The Homey devices all show "unavailable"
When starting the wizard:
![Screenshot from 2023-10-22 19-02-16](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/74005072/e300127f-e5fd-4ea4-b2db-c979aa732f33)

When opening 'manage my existing devices':
![Screenshot from 2023-10-22 19-02-35](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/74005072/d30cd8cb-e394-4a4d-907b-e2f5dd52b8cd)

Console.re stays completely empty, the app doesn't seem to be running.
I created a diagnostic report as well: 
1b058edb-5dd2-44b1-93f9-a0b543d18810

**To Reproduce**
Not shure, I have only one Homey to test with

**Expected behavior**
a working app ;-) 

**Desktop (please complete the following information):**
- Linux Mint 21.2 cinnamon
- Chrome v. 118.0.5993.88 (Official Build) unknown (64-bit)

**Smartphone (please complete the following information):**
n.a.

**Additional context**
Console.re stays completely empty (yes I enabled it)

**Homey**
- Pro 2019 v8.1.6
- ESPHome app v.1.1.30
- same behaviour as the live version, both fail

Thanks in advance for looking into it

**Comment by Baldhor** [comment]
@PeterKawa I changed title of the bug, there are currently 292 devices connectedf using this app :)
Let's start thinking your problem is "specific" :)

Can you collect the debug info from the manage your physical page ? or you cannot connect at all ?

**Comment by Baldhor** [comment]
@PeterKawa ok so I took more time to read, and I agree it looks like a big fail here ...

From what I understand:
1/ You already have 4 virtual devices created with ESPhome app
2/ All of them show as unavailable
3/ The Wizard fail big time
4/ console.re fail big time too ...

Which version of the app you currently use ?
![image](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/57640917/c0e66dd3-4676-44ed-bb8b-df7649084236)

**Comment by Baldhor** [comment]
I'm analyzinzing your diag report:
1/ You are on 1.1.30 (last test version)
```
    manifest: {
      _comment: 'This file is generated. Please edit .homeycompose/app.json instead.',
      id: 'nl.inversion.esphome',
      version: '1.1.30',
```

2/ The root cause is probably this: "Error: Handshake failure: Handshake MAC failure"
> Which I supposed is related to the encryptionKey

> Did you change it lately???

3/ Still the wizard should be stable in its behaviour and allow you to modify your encryptionKey
> I will look into that part

**Comment by Baldhor** [comment]
> I will look into that part

I found a bug introduced with the function "collect debug info", and I pushed a fix in test version.

@PeterKawa It still doesn't explain why the app cannot connect to your devices!
> Seems only the "Presence Badkamer nw" device has a handshake issue, the others just cannot connect (Error: connect ECONNREFUSED 192.168.1.23:6053)!

Sidenote, I have no idea where you took those screenshot. For me the browser view of ESPhome device looks like:
![image](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/57640917/b6fb73f9-2808-45f9-abd7-1aee08d31a79)

You are the only user I see with a different browser experience.

**Comment by Baldhor** [comment]
@PeterKawa Maybe you can reach me on slack on by mails, I feel like your issue is very specific, and I believe not related to the app (but I cannot exclude it yet)

**Comment by Baldhor** [comment]
remark: you should now be able to modify the device password and encruptionKey ... Is it the issue?

**Comment by PeterKawa** [comment]
Thanks, for the swift actions Baldhor,
The title is not *I cannot find my devices* 
It did not **show** any of my devices, which to me, is something completely different! So I changed the title again.

With v1.1.31 it now shows both my physical and virtual devices again and all's seem to work again

I can now add a physical device again
And I can add a virtual device based on the physical one.

I also can edit my existing physical devices again, this afternoon **they simply did not appear** in the wizard, like you could see at the wizard screenshots.

But why they were unreachable is another thing, they all had received other IP's during the time I wasn't busy with ESPHome
Now I could adjust the IP's and all's up and running again.
This has by the way nothing to do with the app showing none of my physical and virtual devices at all.


Collected debug info
{
  "versions": {
    "platform": "local",
    "platformVersion": 1,
    "version": "8.1.6",
    "appVersion": "1.1.31"
  },
  "physicalDeviceId": "migrated4",
  "physicalDevice": {
    "physicalDeviceId": "migrated4",
    "status": "unavailable",
    "used": true,
    "name": "Presence Badkamer nw",
    "ipAddress": "192.168.1.26",
    "port": "6053",
    "encryptionKey": "<hidden value>",
    "password": "<no value>",
    "nativeCapabilities": [],
    "rawData": []
  }
}

**Comment by PeterKawa** [comment]
> Sidenote, I have no idea where you took those screenshot. For me the browser view of ESPhome device looks like:
That screenshot is the homescreen of ESPHome when you run it locally. Similar looking screen is to be seen in the ESPHome integration of HomeAssistant.
Of course I have device pages as well:
![Screenshot from 2023-10-23 03-02-51](https://github.com/Baldhor/Homey-ESPhome-Enhanced/assets/74005072/84ddfcea-3b43-4a28-b11c-18fa58953fd9)

**Comment by PeterKawa** [comment]
> From what I understand:
1/ You already have 4 virtual devices created with ESPhome app
2/ All of them show as unavailable
3/ The Wizard fail big time
4/ console.re fail big time too ...
Which version of the app you currently use ?

1/ correct
2/ no, they did not show at all. There were no devices to be seen, nothing, empty.
3/ Yes I could not do anything while there were no devices shown. Also I could not add new devices, the 'wait' circle kept going for 15 minutes before I killed it.
4/ Yes, no rule was added. Zero. None

App version was in my report. When the live version did not work for me, I installed v1.1.30 test, but with same behaviour
>Homey
Pro 2019 v8.1.6
ESPHome app v.1.1.30
same behaviour as the live version, both fail
